sklearn/tests/test_calibration.py

```diff
@@ -996,3 +998,87 @@ def fit(self, X, y, sample_weight=None, fit_param=None):
     CalibratedClassifierCV(estimator=TestClassifier()).fit(
         *data, fit_param=np.ones(len(data[1]) + 1)
     )
+
+
+def test_calibrated_classifier_cv_works_with_large_confidence_scores(
+    global_random_seed,
+):
+    """Test that :class:`CalibratedClassifierCV` works with large confidence
+    scores when using the `sigmoid` method, particularly with the
+    :class:`SGDClassifier`.
+
+    Non-regression test for issue #26766.
+    """
+    prob = 0.67
+    n = 1000
+    random_noise = np.random.default_rng(global_random_seed).normal(size=n)
+
+    y = np.array([1] * int(n * prob) + [0] * (n - int(n * prob)))
+    X = 1e5 * y.reshape((-1, 1)) + random_noise
+
+    # Check that the decision function of SGDClassifier produces predicted
+    # values that are quite large, for the data under consideration.
+    cv = check_cv(cv=None, y=y, classifier=True)
+    indices = cv.split(X, y)
+    for train, test in indices:
+        X_train, y_train = X[train], y[train]
+        X_test = X[test]
+        sgd_clf = SGDClassifier(loss="squared_hinge", random_state=global_random_seed)
+        sgd_clf.fit(X_train, y_train)
+        predictions = sgd_clf.decision_function(X_test)
+        assert (predictions > 1e4).any()
+
+    # Compare the CalibratedClassifierCV using the sigmoid method with the
+    # CalibratedClassifierCV using the isotonic method. The isotonic method
+    # is used for comparison because it is numerically stable.
+    clf_sigmoid = CalibratedClassifierCV(
+        SGDClassifier(loss="squared_hinge", random_state=global_random_seed),
+        method="sigmoid",
+    )
+    score_sigmoid = cross_val_score(clf_sigmoid, X, y, scoring="roc_auc")
+
+    # The isotonic method is used for comparison because it is numerically
+    # stable.
+    clf_isotonic = CalibratedClassifierCV(
+        SGDClassifier(loss="squared_hinge", random_state=global_random_seed),
+        method="isotonic",
+    )
+    score_isotonic = cross_val_score(clf_isotonic, X, y, scoring="roc_auc")
+
+    # The AUC score should be the same because it is invariant under
+    # strictly monotonic conditions
+    assert_allclose(score_sigmoid, score_isotonic)
+
+
+def test_sigmoid_calibration_max_abs_prediction_threshold(global_random_seed):
+    random_state = np.random.RandomState(seed=global_random_seed)
+    n = 100
+    y = random_state.randint(0, 2, size=n)
+
+    # Check that for small predictions ranging from -1 to 1, the threshold
+    # value has no impact on the outcome
+    predictions_small = random_state.uniform(low=-1, high=1, size=100)
```

Comment:
Actually we need a range larger than `[-1, 1]` (e.g, `[-2, 2]`) to make sure that the scale derived from `max(abs(predictions))` when below the threshold is significantly distinct from the unit scale.

Reply:
Then if we need to keep the 
```
AB_ /= scale_constant
```
If I set threshold to 0.1 it will differ from the case where threshold is 10 or 30 because of the scaling.

Reply:
Yes indeed, we need to scale back the trained parameters to match the scale of the original problem of interest.

Reply:
Then we keep the scaling. But how do you suggest we handle the test to check thresholds don't matter for predictions [-2, 2]? Because here if we set threshold to 0.1 it scales and then rescales and produces different values when no scaling is applied when threshold is 10 or 30. The difference in the coeff values occur precisely due to the extra scaling factor we divide by in the first case.

Reply:
I did some investigation and I think what were the problems:

- we need to only re-scale the multiplicative parameter, not the offset which is only dependent on the scale of the outcome variable;
- we need to avoid in-place modifications of the input predictions are they are subsequently used in the caller;
- we need to adjust the assertions tolerance to match the default tolerance of the underlying solver (lbfgs).

Reply:
@ogrisel Thank you very much.